### PR TITLE
MEP_oM: Added CameraDevice defintion

### LINKED ISSUE #1078: MEP_oM: Create definition for Camera devices
#### Description:
Create the object definition for Camera devices, that follows some internal properties from MEP elements.
It should have as basic properties things such as:

-  Start and end point
-  The planarized cone geometry
-  Installation height

### PR DESCRIPTION
**Pull request by tiagogrossi**
### Issues addressed by this PR

Closes #1078 

Added the definition for CameraDevice.

### Test files
Test file [here](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/BHoM/MEP_oM/Issue1078-Create%20definition%20for%20Camera%20devices/MEP_oM-Issue1078-Create%20definition%20for%20Camera%20devices.gh?csf=1&web=1&e=CERKJ1).

### Changelog
Added new object `BH.oM.MEP.Fixtures.CameraDevice`
Added new enums `BH.oM.MEP.Enums.CameraType`
Added new enums `BH.oM.MEP.Enums.MountingType`

### CODE REVIEW

**Comment by tiagogrossi** [comment]
@kayleighhoude I think I followed the nomenclature we determined before, but could you please check that you're happy with this? I've put it under fixtures, but opted to call it a `CameraDevice` instead of `CameraFixture`. What do you think?

File: MEP_oM/Enums/CameraType.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    [Description("A camera type that describes its basic characteristcs.")]
+    public enum CameraType
+    {
+        Domed,
+        Covert,
+        Bullet,
+        Fisheye,
+        MultiMegapixels,
+        PanTiltZoom,
+        Undefined
```
Comment (FraserGreenroyd) [inline MEP_oM/Enums/CameraType.cs:38]: Enum should be in alphabetical order, with `Undefined` as the first option

File: MEP_oM/Enums/MountingType.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    public enum MountingType
+    {
+        Wall,
+        Soffit,
+        Ceiling,
+        Pendant,
+        Column, 
+        Floor,
+        Facade,
+        Undefined
```
Comment (FraserGreenroyd) [inline MEP_oM/Enums/MountingType.cs:39]: Enum should be alphabetical with `Undefined` as the first option.

File: MEP_oM/Fixtures/CameraDevice.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        [Description("The Camera mounting type that describes how it's installed.")]
+        public MountingType Mounting { get; set; } = MountingType.Undefined;
+
+        [Description("The Camera type that describes its characteristics.")]
+        public CameraType Type { get; set; } = CameraType.Undefined;        
+
+        [Description("The Camera megapixels that determines its image quality.")]
+        public double Megapixels { get; set; } = 0;
+
```
Comment (FraserGreenroyd) [inline MEP_oM/Fixtures/CameraDevice.cs:59]: All properties must be `virtual`

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by enarhi** [comment]
Closing, PR moved to #1091

**Comment by FraserGreenroyd** [comment]
@enarhi please also handle the Engine PR linked to this as appropriate.

**Pull request closed without merging**